### ISSUE TITLE
Docker builder plugin: allow passing in secrets 

### ISSUE THREAD

**Issue by rkrasiuk** [hashicorp/waypoint#542]
**Is your feature request related to a problem? Please describe.**
Support passing build secret information when building docker image. 

**Describe the solution you'd like**
```hcl
build {
  use "docker" {
    buildkit    = false
    disable_entrypoint = false
    secrets = [
      "id=supersecret,src=/path/to/supersecret"
    ]
  }
}
```

**Describe alternatives you've considered**
Alternatively, docker builder could be extended to allow passing arbitrary cli flags.

**Comment by mitchellh** [comment]
Yes we'd like to eventually support variables natively in the Waypoint configs (similar to Terraform). Not there yet!

Re: the original issue, good idea to support.

**Comment by rkrasiuk** [comment]
@mitchellh no rush, you've done a great job already on this! another idea might be extending `exec` plugin to implement build interface, but i'd say that's whole another topic ; -)

**Comment by winmillwill** [comment]
It's worth noting that the secrets capability of buildkit isn't just about parameter passing, it's also about not leaking secrets into the docker registry.

**Comment by mitchellh** [comment]
One issue we just discovered with this is that in the next version of Waypoint we use [Kaniko](https://github.com/GoogleContainerTools/kaniko) to build container images within on-demand runners, but Kaniko doesn't support buildkit secrets.
